app/services/analytics.rb

```diff
@@ -134,13 +134,22 @@ def sp_request_attributes
       [v.name.sub('http://idmanagement.gov/ns/assurance/', ''), true]
     end.to_h
     attributes.reject! { |_key, value| value == false }
+    attributes.merge!(identifier)
     attributes.transform_keys! do |key|
       key.to_s.chomp('?').to_sym
     end
 
     { sp_request: attributes }
   end
 
+  def identifier
+    sp_request_url = session&.dig(:sp, :request_url)
+    return {} if sp_request_url.blank?
+
+    app_identifier = UriService.params(sp_request_url)['app_identifier']
```

Comment:
This approach looks good! I vote however for a name that includes the "login_gov" somewhere to help clarify the meaning

Reply:
I don't mean to add further complexity, but references to an `app_id` are a potential collision with the `app_id` used in the `service_providers` table: https://github.com/18F/identity-idp-config/blob/91dfc9b0b3298db9bbbc25c3e33c94a542761d06/lib/models/service_provider.rb#L56

Reply:
`login_gov_app_differentiator` kinda verbose, but better to be explicit if there is a chance for confusion

Reply:
`sub_app` ?